projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -250,9 +256,11 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent> {
         }
     }
 
-    //  TODO: refactor update_cell. Maybe separate logic in two methods - one with transaction
-    //  and one without transaction
-    public update_cell(id: string, rowID, columnID, editValue) {
+    public get_grid_edit_state(id: string, rowID, columnID, editValue): {
```

Comment:
@rkaraivanov @damyanpetev 
`update_cell` method in the API is not responsible for calling `escape_cell_inEditMode`, but cancelling `onCellEditDone` should leave the cell in edit mode (**?**), so I've created a method to get the current 'edit state' and I'm using it to pass it from wherever `update_cell` is called, so that the event can be cancelled properly when called either through the API or as a result of UI interactions.
When you can, please take a look and let me know if this seems okay to you.

Reply:
I'd personally prefer if `update_cell` did just that - simple update and the control to be before that, but that's a talk for a future time.
So as long as you name it like what it does (get edit event args or something) the method is fine